### PR TITLE
[Android] Bump target SDK version in Android manifest

### PR DESCRIPTION
**Pull request by simonrozsival**
We should run our tests targeting API 31 which is currently the [Google Play's requirement](https://developer.android.com/google/play/requirements/target-sdk).

It will be necessary to use Android SDK 31+ for local builds. The build process will show an error message instructing the developer to install newer build tools it they're out of date. The CI already uses SDK version 33 (https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/713).

### CODE REVIEW

**Review by mdh1418 — APPROVED** [review]

File: docs/workflow/testing/libraries/testing-android.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ Android SDK and NDK can be automatically installed via the following script:
 set -e
 
 NDK_VER=r23c
-SDK_VER=6200805_latest
-SDK_API_LEVEL=29
-SDK_BUILD_TOOLS=29.0.3
+SDK_VER=8092744_latest
+SDK_API_LEVEL=33
+SDK_BUILD_TOOLS=33.0.0
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply docs/workflow/testing/libraries/testing-android.md:28]: I was matching what we already use in CI: https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/713. But you're right, there's no good reason to put the outdated version into these docs. There's no real benefit of using the same SDK in CI and locally.

File: src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs [hunk 2]
```diff
@@ -302,6 +302,7 @@ public void ProcessStart_UseShellExecute_OnWindows_DoesNotThrow(bool isFolder)
         [ActiveIssue("https://github.com/dotnet/runtime/issues/34685", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
         [InlineData(true), InlineData(false)]
         [SkipOnPlatform(TestPlatforms.iOS | TestPlatforms.tvOS, "Not supported on iOS and tvOS.")]
+        [SkipOnPlatform(TestPlatforms.Android, "Permission denied")]
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs]: `Process.Start` doesn't work when we're attempting to run a shell script.

**Comment by simonrozsival** [comment]
The test `System.Net.Security.Tests.CertificateValidationRemoteServer.ConnectWithRevocation_WithCallback(checkRevocation: True)` consistently fails on physical devices but passes on emulators after the bump. That's quite interesting.

**Comment by simonrozsival** [comment]
The remaining failing tests on Android are related to https://github.com/dotnet/runtime/pull/80909.

**Review by steveisok — APPROVED** [review]

**Pull request merged**